src/utils/query_preprocessing.py

```
@@ -0,0 +1,194 @@
+                messages=[{
+                    "role": "system",
+                    "content": "You are a search query optimizer. Always return valid JSON."
+                }, {
+                    "role": "user",
+                    "content": prompt
+                }],
+                temperature=0,
+                response_format={"type": "json_object"}
+            )
+            
+            # Parse the response
+            result = json.loads(response.choices[0].message.content)
+            return SearchPlan.from_dict(result)
+            
+        except Exception as e:
+            if retry < max_retries - 1:
+                print(f"Error planning search strategy (attempt {retry + 1}/{max_retries}): {e}")
+                time.sleep(retry_delay)
+                retry_delay *= 2  # Exponential backoff
+            else:
+                print(f"Failed to plan search strategy after {max_retries} attempts: {e}")
+                # Fallback to simple strategy
+                return SearchPlan(
+                    search_terms=[query],
+                    keywords=query.split(),
+                    semantic_query=query,
+                    search_breadth="medium"
+                )
+
+
+def plan_search_strategy(query: str, use_cache: bool = True) -> SearchPlan:
+    """
+    Use an LLM to intelligently plan how to search for information.
+    
+    This function analyzes the user's query and generates an optimized
+    search strategy that includes:
+    - Exact phrases to search for
+    - Individual keywords for keyword matching
+    - A reformulated query optimized for semantic search
+    - Search breadth recommendation
+    
+    Args:
+        query: The user's search query
+        use_cache: Whether to cache results for repeated queries
+        
+    Returns:
+        SearchPlan with optimized search strategies
+    """
+    # Check if LLM query planning is enabled
+    if not os.getenv("OPENAI_API_KEY"):
```

This check should be happening in the `get_openai_client` method I suggested above. We should be raising an error, not failing silently too.